server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -3475,8 +3475,17 @@ private void checkAborted() {
                     }
                 };
                 final String partName = fileInfo.partName(i);
-                logger.trace(() -> new ParameterizedMessage("[{}] Writing [{}] to [{}]", metadata.name(), partName, shardContainer.path()));
+                logger.trace("[{}] Writing [{}] to [{}]", metadata.name(), partName, shardContainer.path());
+                final long startMS = threadPool.relativeTimeInMillis();
                 shardContainer.writeBlob(partName, inputStream, partBytes, false);
+                logger.trace(
+                    "[{}] Writing [{}] of size [{}b] to [{}] took [{}ms]",
+                    metadata.name(),
+                    partName,
+                    partBytes,
+                    shardContainer.path(),
+                    threadPool.relativeTimeInMillis() - startMS
```

Comment:
a true nit, but I prefer to capture the `threadPool.relativeTimeInMillis()` explicitly right after the write to not have to worry about any time spent in any of the (obviously fast) method calls here.

Reply:
:) I think here you don't have any guarantees on the order of invocations anyway after JIT (which will inline the method calls anyway) so I'll leave it as is to save a line.